src/storage/src/hummock/mod.rs

```diff
@@ -271,6 +274,18 @@ impl HummockStorage {
     pub fn get_pinned_version(&self) -> PinnedVersion {
         self.pinned_version.load().deref().deref().clone()
     }
+
+    pub fn need_write_throttling(&self) -> bool {
+        self.pinned_version
+            .load()
+            .version_ref()
+            .levels
+            .values()
+            .any(|levels| {
+                levels.l0.as_ref().unwrap().sub_levels.len()
+                    > self.options.throttle_l0_sub_level_number as usize
```

Comment:
Shall we explain here the problem of deadlock when throttle_l0_sub_level_number < level0_tier_compact_file_number?